### PR TITLE
Add wait_block_finalization during signatures update

### PR DESCRIPTION
**Pull request by evgeny-stakewise**

### CODE REVIEW

File: src/commands/start.py [hunk 1]
```diff
@@ -201,6 +203,28 @@ def start(
         log_verbose(e)
 
 
+async def update_exit_signatures_periodically(keystores: Keystores):
```
[the remark this answers is not in the document]
Comment (evgeny-stakewise) [reply src/commands/start.py:209]: Function update_exit_signatures already exists

[the remark this answers is not in the document]
Comment (cyc60) [reply src/commands/start.py:209]: move to exits/tasks.py

File: src/exits/tasks.py [hunk 2]
```diff
@@ -20,43 +29,109 @@
...
 
     async with aiohttp.ClientSession() as session:
         async with session.get(url=url) as response:
             response.raise_for_status()
             data = await response.json()
-    return [x.get('index') for x in data]
+    return data
+
+
+async def _fetch_outdated_indexes(oracle_endpoint: str) -> list[int]:
```
Comment (tsudmi) [inline src/exits/tasks.py:124]: Only 2 line function, maybe not necessary to separate it.

File: src/commands/start.py [hunk 3]
```diff
@@ -201,6 +206,32 @@ def start(
...
+            if outdated_indexes:
+                await update_exit_signatures(keystores, oracles, outdated_indexes)
+
+                # Wait all oracles sync.
+                await wait_oracles_signature_update(oracles)
+        except Exception as e:
+            logger.exception(e)
+
+        elapsed = time.time() - timer_start
+        await asyncio.sleep(float(settings.network_config.SECONDS_PER_BLOCK) - elapsed)
```
[the remark this answers is not in the document]
Comment (evgeny-stakewise) [reply src/commands/start.py:232]: `asyncio.sleep` can handle negative time. It sleeps for 0 sec.

File: src/common/contracts.py [hunk 4]
```diff
@@ -112,6 +112,25 @@ async def get_last_rewards_update(self) -> RewardVoteInfo | None:
         )
         return voting_info
 
+    async def get_exit_signatures_updated_events(
+        self, from_block: BlockNumber, to_block: BlockNumber
+    ) -> list[EventData]:
+        return await self.events.ExitSignaturesUpdated.get_logs(
```
[the remark this answers is not in the document]
Comment (evgeny-stakewise) [reply src/common/contracts.py]: Added

**Review by cyc60 — APPROVED** [review]

**Pull request merged**